src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs

```diff
@@ -38,27 +38,11 @@ private enum InvertIfStyle
         public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
             var (document, textSpan, cancellationToken) = context;
-            if (!textSpan.IsEmpty)
-            {
-                return;
-            }
-
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(textSpan.Start);
 
-            var ifNode = token.GetAncestor<TIfStatementSyntax>();
-            if (ifNode == null)
-            {
-                return;
-            }
-
-            if (ifNode.OverlapsHiddenPosition(cancellationToken))
```

Comment:
Wondering if this check should also move to helper, not sure if we have any refactorings that would want to be enabled if the selected node overlaps hidden position or preprocessor directives. @CyrusNajmabadi ?

Regardless, even if this was desired, it can come in a separate PR.

Reply:
Good question and while we're discussing that there're some refactorings that check the type of File. We might want to introduce that as well (TBH I'm not sure what the exact semantics of either of these checks are).

Reply:
yeah, i have a feeling this could just be a chekc that applies to 100% of features that use hte new helper service.  can we at least track this?

Reply:
What exactly does the check OverlapsHiddenPosition does?

Reply:
I mean, imagine you have a large local function, you want to make it static but it has a collapsed region inside. Would the check prevent that?

Reply:
It's not about collapsed regions :)

it's about regions that users can't actually get to (i.e. in asp.net/razor scenarios).  IN those cases, the code will actually be something like:

```c#
void HiddenMethod()
{
     // asp goop

// ----- user can edit from this point on

// ---- user can not edit from this point on

     // asp goop
}
```

Things only overlap hidden regions if they're actually constructs that don't really belong to the user, but instead belong primarily to asp/razor.  In that case, we really don't ever expect hte user to be able to edit that stuff.  Note, this only affects things that *overlap*.  i.e. had their start (and usually their end) be in the hidden section.  In the example, that would only be the HiddenMehtod, and the statements in those hidden sections.  Statements in the middle would be fine.

Reply:
Ok, so if a node OverlapsHiddenPosition is guaranteed to start at a location that isn't owned by a user. If that's the case then I can't se an example of _not_ ever wanting having this check.

Reply:
What about these checks, what's their semantics?
```csharp
            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
            {
                return;
            }
```

Reply:
that one is less clear.  basically syntactic things are totally safe.  we'd have to enumerate semantic things.

That said, thse are all refactorings... so i'm struggling to think why we wouldn't want htem in the misc workspace.  It's not like they're analyzers, or things that show stuff in the UI otherwise.

I think these were all from a very long time ago when misc workspace wasn't well supported, and whne refactorings/analyzers were the same thing.  I think we could remove tehm all (only from refactorings), but it would likely be best to do that in a single PR that we can review.

Reply:
Sure, what exactly is `MiscellaneousFiles` workspace? I mean, what should I imagine under that term?

Reply:
When you open a random .cs or .vb file *outside* of a project :)

Reply:
Looks like this in the IDE:

![image](https://user-images.githubusercontent.com/4564579/61502092-3ec09b80-a987-11e9-8282-9bc5e8125cb3.png)